app/code/core/Mage/Directory/Model/Observer.php

```diff
@@ -22,49 +22,57 @@
 class Mage_Directory_Model_Observer
 {
     public const CRON_STRING_PATH = 'crontab/jobs/currency_rates_update/schedule/cron_expr';
-    public const IMPORT_ENABLE = 'currency/import/enabled';
-    public const IMPORT_SERVICE = 'currency/import/service';
+    public const IMPORT_ENABLE    = 'currency/import/enabled';
+    public const IMPORT_SERVICE   = 'currency/import/service';
 
-    public const XML_PATH_ERROR_TEMPLATE = 'currency/import/error_email_template';
-    public const XML_PATH_ERROR_IDENTITY = 'currency/import/error_email_identity';
+    public const XML_PATH_ERROR_TEMPLATE  = 'currency/import/error_email_template';
+    public const XML_PATH_ERROR_IDENTITY  = 'currency/import/error_email_identity';
     public const XML_PATH_ERROR_RECIPIENT = 'currency/import/error_email';
 
     /**
+     * @param Mage_Cron_Model_Schedule|null $cron
      * @throws Mage_Core_Exception
      */
-    public function scheduledUpdateCurrencyRates()
+    public function scheduledUpdateCurrencyRates($cron = null)
```

Comment:
Pls dont signature of used methods. Better implement a new one,

Reply:
How about this?
```php
if (func_num_args() && func_get_args()[0] instanceof Mage_Cron_Model_Schedule) {
    $schedule = func_get_args()[0];
    //....
}

Reply:
I'd add a new method w/ parameter and all logic in and forward to this method. Similiar i did with #2940.